pkgs/development/python-modules/pubnub/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, aiohttp
+, buildPythonPackage
+, cbor2
+, fetchFromGitHub
+, pycryptodomex
+, pytestCheckHook
+, pyyaml
+, pytest-vcr
+, pytest-asyncio
+, requests
+, six
+}:
+
+buildPythonPackage rec {
+  pname = "pubnub";
+  version = "4.8.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = "python";
+    rev = "v${version}";
+    sha256 = "16wjal95042kh5fxhvji0rwmw892pacqcnyms520mw15wcwilqir";
+  };
+
+  propagatedBuildInputs = [
+    cbor2
+    pycryptodomex
+    requests
+    six
+  ];
+
+  checkInputs = [
+    aiohttp
+    pycryptodomex
+    pytest-asyncio
+    pytestCheckHook
+    pytest-vcr
+
+  ];
+
+  # Some tests don't pass with recent releases of tornado/twisted
```

Comment:
Is this a test issue or a problem with the code itself? Maybe we should require an older version to not ship broken packages.

Reply:
from the comment, it sounds like the tornado updates broke the test suite.

This is a wide spread issue within the python ecosystem in general.

Generally upstreams will just pin tornado or other dependencies to small range, at the cost of being potentially incompatible with other package installations